rfd/0199-teleport-key-agent.md

```diff
@@ -0,0 +1,514 @@
+| `teleport.agent` | `false` | Runs the Teleport Key Agent |
+
+**future work**
+
+The Teleport Connect developers have tested with linking login storage with
+a [symlink](https://github.com/gravitational/teleport/issues/25806), which is
+essentially the same as changing Connect's Teleport home directory. Without any
+additional changes, this testing resulted in some state drift and other errors.
+
+These issues are expected to impact [signing mode](#signing-mode) which won't
+offer sophisticated login syncing logic. [Login sync mode](#login-sync-mode-future-work),
+which is detailed as *future work*, could be implemented to mitigate state drift
+issues in Teleport Connect.
+
+Furthermore, once we develop a reliable framework for login state syncing, whether
+it's with [the login sync agent mode](#login-sync-mode-future-work) or otherwise, we may consider
+defaulting the config values above to `~/.tsh` and `true` respectively.
+
+##### `tsh agent`
+
+Teleport Connect is the primary method for running the Teleport Key Agent as it
+provides the best UX as a naturally long-lived process and UI components for
+Hardware Key pin/touch prompts. However there are likely to be some scenarios
+where running Teleport Connect is not an option. Therefore we will make
+`tsh agent` available as well.
+
+The Teleport Key Agent can be launched explicitly with `tsh agent`. This will
+serve the agent as long as it remains running. The user will be prompted to
+re-login as needed, both at the start and as needed when their certs expire.
+
+Note: `tsh agent` will also be a lightweight command useful for setting the
+groundwork with the slightly more extensive Teleport Connect implementation.
+It will definitely speed up my testing of the feature, so at the very least
+we could make this a hidden command.
+
+#### Agent already running
+
+If either Teleport Connect or `tsh agent` finds that there is already an active
+Teleport Key Agent for the same Teleport home directory, an error will be
+returned.
+
+```console
+> tsh agent
+Error: another instance of Teleport Key Agent is already running
+```
+
+In the case of Teleport Connect, the error will be displayed in the UI but will
+not impact it from running.
+
+#### Agent stops running
+
+If the agent stops unexpectedly during a dependent client's operation, it may
+lead to an error for the dependent client. If the client is able to re-login
+to fix the error, the client should do so, switching from key agent storage
+to file storage.
+
+In cases where the agent's login state was fully in file storage, the dependent
+client can change sources and continue without the need to re-login.
+
+#### Prompts
+
+The agent client is responsible for any prompts that occur within the key
+storage and signing interface. Currently, this is limited to Hardware Key
+pin/touch prompts, but there are some additional use cases we can explore.
```

Comment:
So, if I use hardware keys, I'll get an error in tsh if I don't have an agent? Or is the agent optional for better UX?